src/scenic/core/scenarios.py

```diff
@@ -1,518 +1,595 @@
+                        continue
+                    if oi.intersects(oj):
+                        oi_name = oi.name if hasattr(oi, 'name') else "Object"
+                        oj_name = oj.name if hasattr(oj, 'name') else "object"
+
+                        raise InvalidScenarioError(f'{oi_name} at {oi.position} intersects'
+                                                   f' {oj_name} at {oj.position}')
+
+    def generate(self, maxIterations=2000, verbosity=0, feedback=None):
+        """Sample a `Scene` from this scenario.
+
+        For a description of how scene generation is done, see `scene generation`.
+
+        Args:
+            maxIterations (int): Maximum number of rejection sampling iterations.
+            verbosity (int): Verbosity level.
+            feedback (float): Feedback to pass to external samplers doing active sampling.
+                See :mod:`scenic.core.external_params`.
+
+        Returns:
+            A pair with the sampled `Scene` and the number of iterations used.
+
+        Raises:
+            `RejectionException`: if no valid sample is found in **maxIterations** iterations.
+        """
+        objects = self.objects
+
+        # choose which custom requirements will be enforced for this sample
+        activeReqs = [req for req in self.initialRequirements if random.random() <= req.prob]
+
+        # do rejection sampling until requirements are satisfied
+        rejection = True
+        iterations = 0
+        while rejection is not None:
+
+            if iterations > 0:  # rejected the last sample
+                if verbosity >= 2:
+                    print(f'  Rejected sample {iterations} because of: {rejection}')
+                if self.externalSampler is not None:
+                    feedback = self.externalSampler.rejectionFeedback
+            if iterations >= maxIterations:
+                raise RejectionException(f'failed to generate scenario in {iterations} iterations')
+            iterations += 1
+            try:
+                if self.externalSampler is not None:
+                    self.externalSampler.sample(feedback)
+                sample = Samplable.sampleAll(self.dependencies)
+            except RejectionException as e:
+                optionallyDebugRejection(e)
+                rejection = e
+                continue
+            rejection = None
+            ego = sample[self.egoObject]
+
+            # Normalize types of some built-in properties
+            for obj in objects:
+                sampledObj = sample[obj]
+                assert not needsSampling(sampledObj)
+                # position, heading
+                assert isinstance(sampledObj.position, Vector)
+                assert isinstance(sampledObj.heading, float)
+                # behavior
+                behavior = sampledObj.behavior
+                if behavior is not None and not isinstance(behavior, Behavior):
+                    raise InvalidScenarioError(
```

Comment:
I think this is unreachable since we normalize `behavior` in `Object._specify` (but good to check).

Reply:
I kept the needsSampling check since that seems like it might come in handy.